internal/cri/config/config.go

```diff
@@ -291,7 +291,7 @@ type ImageConfig struct {
 	//   "base": "docker.io/library/ubuntu:latest"
 	// Migrated from:
 	// (PluginConfig).SandboxImage string `toml:"sandbox_image" json:"sandboxImage"`
-	PinnedImages map[string]string `toml:"pinned_images" json:"pinned_images"`
+	PinnedImages map[string]string `toml:"pinned_images" json:"pinnedImages"`
```

Comment:
Curious; will this fix the linked issue? Because I see that  mentions code that looks for (commented above) `sandBoxImage`, not `pinnedImages`  (plural); https://github.com/kubernetes/kubernetes/blob/8f8c94a04d00e59d286fe4387197bc62c6a4f374/cmd/kubeadm/app/preflight/checks.go#L841C2-L848C3

Reply:
not the same issue.

Just find this potential typo during looking into that issue.

Reply:
I think the issue is related to https://github.com/containerd/containerd/pull/11114 and https://github.com/containerd/containerd/issues/11117

Reply:
sandboxImage vs. pinnedImages is handled in this configuration migration:

```
        var pinnedImages map[string]interface{}
        if v, ok := dst["pinned_images"]; ok {
                pinnedImages = v.(map[string]interface{})
        } else {
                pinnedImages = map[string]interface{}{}
        }

        if simage, ok := src["sandbox_image"]; ok {
                pinnedImages["sandbox"] = simage
        }
        if len(pinnedImages) > 0 {
                dst["pinned_images"] = pinnedImages
        }
```

when it moved from the RuntimeConfig to the ImageConfig, in 3baf5edb8b5b3342c1d729ae857581865cb43b21

Reply:
> maybe it needs to delete the sandbox_image, during the above process?

Yeah there are many other fields that also have the warning even after the migration. I have a PR to fix that a while ago https://github.com/containerd/containerd/pull/10980. Need to revisit the PR and see if I can get some review there then.

Reply:
@djdongjin  https://github.com/containerd/containerd/pull/10980 was merged.

@thaJeztah  Could you take a look and see if the current PR is meaningful?